### PR TITLE
samples: bluetooth: peripheral_hids_mouse: Don't start advertising when active

### PR DESCRIPTION
**Pull request by maje-emb**
Added a check to ensure Bluetooth advertising is not initiated if it is already running.

### CODE REVIEW

**Review by grochu — APPROVED** [review]
The updated advertising logic looks good to me.

File: samples/bluetooth/peripheral_hids_mouse/src/main.c [hunk 1]
```diff
@@ -162,14 +166,23 @@ static void advertising_continue(void)
 
 	if (!k_msgq_get(&bonds_queue, &addr, K_NO_WAIT)) {
 		char addr_buf[BT_ADDR_LE_STR_LEN];
+		int err;
+
+		if (atomic_test_bit(&adv_state, ADV_RUNNING)) {
```
Comment (KAGA164) [inline samples/bluetooth/peripheral_hids_mouse/src/main.c:171]: I would say that we do not need atomic here. Every access to adv_state seems to be done from the same no preemptive context

Comment (maje-emb) [reply samples/bluetooth/peripheral_hids_mouse/src/main.c:171]: `adv_state` is used in the connected and disconnected callbacks. As far as I know, these callbacks are called from different threads, BT and sysworkq.

Comment (KAGA164) [reply samples/bluetooth/peripheral_hids_mouse/src/main.c:171]: Yes and they are cooperative threads. You also call advertising_continue only from sysworkq

**Review by doki-nordic — APPROVED** [review]

File: samples/bluetooth/peripheral_hids_mouse/src/main.c (hunk 1 above)
Comment (maje-emb) [reply samples/bluetooth/peripheral_hids_mouse/src/main.c]: Right. Changed to bool.

**Review by KAGA164 — APPROVED** [review]

**Pull request merged**